shared/src/fns.rs

```diff
@@ -5,12 +5,45 @@ use tokio::fs;
 use crate::Url;
 
 pub fn expand_path(p: impl AsRef<Path>) -> PathBuf {
-	let p = p.as_ref();
+	let mut p = p.as_ref();
+
+	// expand the environment variable by calling the "echo" command, in linux case,
+	// this also expands the '~' path
+	#[cfg(target_os = "windows")]
+	let expanded_path = match std::process::Command::new("cmd").args(&["/C", "echo"]).arg(p).output() {
+		Ok(output) if output.status.success() => Some(
+			String::from_utf8_lossy(&output.stdout)
+				.trim_end()
+				.trim_matches('"')
+				.replace("\\\"", "\"")
```

Comment:
When do quote marks appear? I tested several cases but none of them does this.

Reply:
When there is `"` in the file name. You'll see it if you try to run:
```rust
std::process::Command::new("cmd")
        .arg("/C")
        .arg("echo")
        .arg(std::path::Path::new("a\"b.txt"))
        .output();
```

Reply:
```
a\"b.txt
```

Seems the quote marks only appear in the middle, not on both sides. Is `trim_matches('"')` here meaningless?

Reply:
That is for another case where there is space in the file name.

Reply:
Oh I see, thanks